src/main/java/com/google/sps/server/ScheduleCalendar.java

```diff
@@ -74,12 +74,20 @@ public void addSchedule(Schedule schedule) throws IOException {
 
       // for each course in the schedule, add an event to the calendar
       for (ScheduledCourse currentCourse : schedule.getCourses()) {
-        List<TimeRange> sectionTimes = currentCourse.getSection().getMeetingTimes();
-        for (int i = 0; i < sectionTimes.size(); i++) {
-          String startTime = CalendarUtils.calculateStartTime(sectionTimes.get(i), month, year, nextSun);
-          String endTime = CalendarUtils.calculateEndTime(sectionTimes.get(i), month, year, nextSun);
+        List<TimeRange> lectureSectionTimes = currentCourse.getLectureSection().getMeetingTimes();
+        for (int i = 0; i < lectureSectionTimes.size(); i++) {
+          String startTime = CalendarUtils.calculateStartTime(lectureSectionTimes.get(i), month, year, nextSun);
+          String endTime = CalendarUtils.calculateEndTime(lectureSectionTimes.get(i), month, year, nextSun);
           addEvent(currentCourse, startTime, endTime);
         }
```

Comment:
It's best not to repeat the same code twice if you can avoid it. In this case, you could create a private helper function that adds a Section to the calendar.

Reply:
Good point, I will make that improvement now, thanks!

Reply:
Instead of adding a helper function that then called addEvent(), or returning both the startTime and endTime to addSchedule() and then calling addEvent(), I just added the two startTime / endTime calculations to addEvent(), because this still avoids the duplicate code and seemed simplest. If a helper function would still be clearer, though, let me know and I'll make the change!

Reply:
That solves the problem of the startTime and endTime being duplicated, but you still have two for-loops. It also makes the parameters being passed into `addEvent` a little more difficult to understand.

I'd recommend splitting out a separate function that contains the for-loop and calls `addEvent` with the previous parameters. Something like the following might work:
`private void addSectionToCalendar(Section section, java.util.Calendar calendar) { ... }`

Also something to consider, but don't necessarily need to change now: Your calculateStartTime and calculateEndTime in CalendarUtils could take a java.util.Calendar (instead of month, year, and nextSun) and then call the functions getCurrentMonth, getCurrentYear, and getDateOfNextSunday inside your start and end time functions. This would simplify things from a code readability perspective, since this class doesn't need to know the values of month, year, and nextSun.

Reply:
That makes sense, that would be simpler. Thanks for the feedback, I'll work on that.